getting-started/main.cpp

```diff
@@ -161,7 +162,7 @@ static void sign_a_message_using_rsa(const uint8_t *key, size_t key_len)
     }
 
     /* Sign message using the key */
-    status = psa_asymmetric_sign(handle, PSA_ALG_RSA_PKCS1V15_SIGN_RAW,
+    status = psa_asymmetric_sign(handle, alg,
```

Comment:
Is this signing random data on the stack? Shouldn't we provide some example data to be hashed and signed? We should also check if we need to do the hash ourselves first, or if psa_asymmetric_sign() will do the hashing for us; I suspect it does the hashing for us.

Reply:
Very good catch. I just checked the API doc, `hash` is the _input_ and `signature` is the output, and `alg` is the signature algorithm for signing. So this indeed signs random data on the stack...

In principle, `hash` can be any value or even a plain text message. But as a demo, it's nice to actually hash some experimental data then sign it. :)

I'll rework this part.

Reply:
Oh we already have a `hash_a_message()` example. To avoid duplication I won't do hashing again, but will add a comment explaining this.

Reply:
If we set up a new pre-hashed buffer, that'd probably be good. SHA-256 hash of some fun message or whatever.

Reply:
We just need something that realistically looks like a hash for example purposes, but don't need to verify it.

Reply:
Done